NumericalDerivatives/R/numHess.R

```diff
@@ -0,0 +1,35 @@
+#' Numerically Calculate Hessian matrix
+#' @description This function numerically calculates the Hessian matrix of a given function
+#' evaluated at given values of arguments.
+#' @param func The function on which Hessian matrix is calculated.
+#' @param x0 The values of arguments at which the Hessian matrix is evaluated.
+#' @param xTol The tolerance, where a small \code{xTol} corresponds to increased accuracy
+#' of the numerical procedure.
+#' @param ind_rowvar Index of rows where the Hessian matrix is calculated. Default is all rows.
+#' @param ind_colvar Index of columns where the Hessian matrix is calculated. Default is all columns.
+#' @examples
+#' example <- function(x) sqrt(prod(x))
+#' numHess(example, c(1, 2, 3, 4, 5), 1e-4, c(1, 3, 5), c(2, 4))
+#' @export
+#' 
+numHess <- function(func, x0, xTol, ind_rowvar = 1:length(x0), ind_colvar = 1:length(x0)) {
+    f0   <- func(x0)
+    Hess <- matrix(0, length(ind_rowvar), length(ind_colvar))
+    
+    for (i in 1:length(ind_rowvar)) {
+        row <- ind_rowvar[i]
+        for (j in 1:length(ind_colvar)) {

```

Comment:
As for the Jacobian, I'd like to see a more R-style implementation using `sapply` or `lapply` with `do.call(cbind, )`